LineFollower.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * LineFollower.cpp
+ *
+ *  Created on: Oct 15, 2020
+ *      Author: gentov
+ */
+
+#include "LineFollower.h"
+
+LineFollower::LineFollower(DrivingChassis* myChassis){
+     this->robotChassis = myChassis;
+}
+
+void LineFollower::lineFollow(){
+	// line following. if both sensors are on white full steam ahead but timeout eventually.
+	  // we need the timeout cause otherwise
+	  int leftSensorValue = analogRead(LEFT_LINE_SENSOR);
+	  int rightSensorValue = analogRead(RIGHT_LINE_SENSOR);
+	  int leftCorrection = 0;
+	  int rightCorrection = 0;
+
+	  if(leftSensorValue >= ON_BLACK && rightSensorValue>= ON_BLACK)
```

Comment:
this is reducing the analog signal to a on-off, BAD. Use the analog vlaue of the sensor reading to proportionally set velocities.

Reply:
While I agree that this is reducing it to binary (on/off), since this is running quickly over and over again in the state machine, I do not know if making it proportional velocity based on error in analog reading (difference between white and black), would add much value.  That being said, I have a growing list of TODOs for the firmware, to which I have added this comment.  Right now, this is working well, but I can come back to this and make it a proportional controller later in the semester if time permits.

I can send you a video of this working as of right now, if you like.

Reply:
i am starting to think that perhaps the line tracking will not be super useful given the precision in the encoders and the small distances involved. Note how the robot arrices at the end point with the orention all skewed. I think using the sinusoidal interpolation and heading correction with the IMU is a more useful stragity for this applicaiton